### PR TITLE
Change BootNode to use Discv5 instead of Kademlia

### PR DESCRIPTION
**Pull request by nisdas**
Part of #3147

### CODE REVIEW

File: tools/bootnode/bootnode_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		t.Errorf("Length of nodes stored in table is noyt expected. Wanted %d but got %d", 2, len(nodes))
+
+	}
+	if nodes[0].ID != listenerNode2.ID {
+		t.Errorf("Wanted node ID of %s but got %s", listenerNode2.ID, nodes[1].ID)
+	}
+
+	nodes = listener2.Lookup(listenerNode.ID)
+	if len(nodes) != 2 {
+		t.Errorf("Length of nodes stored in table is noyt expected. Wanted %d but got %d", 2, len(nodes))
```
Comment (prestonvanloon) [inline tools/bootnode/bootnode_test.go]:
```suggestion
		t.Errorf("Length of nodes stored in table is not expected. Wanted %d but got %d", 2, len(nodes))
```

File: tools/bootnode/bootnode_test.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		t.Fatal(err)
+	}
+
+	// test that both the nodes have the other peer stored in their local table.
+	listenerNode := listener.Self()
+	listenerNode2 := listener2.Self()
+
+	nodes := listener.Lookup(listenerNode2.ID)
+	if len(nodes) != 2 {
+		t.Errorf("Length of nodes stored in table is noyt expected. Wanted %d but got %d", 2, len(nodes))
```
Comment (prestonvanloon) [inline tools/bootnode/bootnode_test.go]:
```suggestion
		t.Errorf("Length of nodes stored in table is not expected. Wanted %d but got %d", 2, len(nodes))
```

File: tools/bootnode/bootnode.go [hunk 3]
```diff
@@ -46,56 +46,53 @@ func main() {
...
+		log.Fatal(err)
+	}
+	return network
 }
 
-func addPrivateKeyOpt(opts []libp2p.Option) []libp2p.Option {
+func extractPrivateKey() *ecdsa.PrivateKey {
+	var privKey *ecdsa.PrivateKey
+	var err error
 	if *privateKey != "" {
```
Comment (shayzluf) [inline tools/bootnode/bootnode.go]: how are we going to get into this code path?

Comment (nisdas) [reply tools/bootnode/bootnode.go]: @shayzluf this is initalized by a flag, so if a flag is given we can use the given private key 
or just generate one in the other case

Comment (shayzluf) [reply tools/bootnode/bootnode.go]: sg

File: tools/bootnode/bootnode.go [hunk 4]
```diff
@@ -46,56 +46,53 @@ func main() {
...
-	opts := []libp2p.Option{
-		libp2p.ListenAddrs(listen),
-	}
-	opts = addPrivateKeyOpt(opts)
+	privKey := extractPrivateKey()
+	listener := createListener(defaultIP, *port, privKey)
 
-	ctx := context.Background()
+	node := listener.Self()
+	fmt.Printf("Running bootnode: /ip4/%s/udp/%d/discv5/%s\n", node.IP.String(), node.UDP, node.ID.String())
```
Comment (prestonvanloon) [inline tools/bootnode/bootnode.go]: Nice

**Pull request merged**